firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistributionLifecycleNotifier.java

```diff
@@ -80,6 +91,25 @@ Activity getCurrentActivity() {
     }
   }
 
+  // TODO (mannyjimenez) Put a lock on this in the future.
```

Comment:
Oh yeah, we should fix this before submitting. I think you can just wrap this entire method body in a `synchronized (lock) {}`, but then you can't call `addOnActivityResumedListener` and `removeOnActivityResumedListener` since they need the lock. Instead you'll have to inline the `this.onActivityResumedListeners.add/remove` calls.

Reply:
I'm missing a lot of context, but I am curious if we can call `addOnActivityResumedListener` from within a synchronized block, since the lock is supposed to be reentrant ( A reentrant lock allows a thread that is holding a lock to acquire it again.) Did you test this and find deadlock?

Reply:
Good point! That path can be tested and see if deadlock occurs